lib/assessment.js

```diff
@@ -29,7 +45,11 @@ var newArray = function newArray(length, defaultValue) {
 // invoking the argument defaultsFunction
 // defaultsFunction should be passed the index of the array element to be set
 var newArray2 = function newArray2(length, defaultsFunction) {
-
+  array2 = [];
+  for (var i = 0; i < defaultsFunction.length; i++){
```

Comment:
You are counting up here to `defaultsFunction.length` rather than the passed-in `length` parameter. A function's `length` attribute is the number of arguments it takes (it's arity) in case you're curious.